index.js

```diff
@@ -37,6 +37,9 @@ var windowsReservedRe = /^(con|prn|aux|nul|com[0-9]|lpt[0-9])(\..*)?$/i;
 var windowsTrailingRe = /[\. ]+$/;
 
 function sanitize(input, replacement) {
+  if (typeof input !== 'string') {
+    input = input.toString();
```

Comment:
How about we throw an error instead?

```suggestion
    throw new Error('Input must be string');
```

Does this work for you?

Reply:
No, it dose not work. for example if my file name is '/user/bin/app/888, ', parameter 'input' type is decide  'number'-> 888 . but my filename is right.

Reply:
I understand that, but can your application call `.toString()` before calling `sanitize()`?

I think throwing an error when the input is invalid is better than automatically calling `.toString()` for the user. Otherwise if someone passes in an object accidentally (due to a programming error or bug), they might get `"[Object object]"` as a filename without noticing their issue. If instead they get an error `"Error: Input must be string"` they will be more likely to understand the problem and fix it.